soccer/radio/NetworkRadio.cpp

```diff
@@ -24,11 +24,16 @@ void NetworkRadio::startReceive() {
 
 bool NetworkRadio::isOpen() const { return _socket.is_open(); }
 
-void NetworkRadio::send(Packet::RadioTx& packet) {
+void NetworkRadio::send(
+    Packet::RadioTx& radioTx,
+    const std::array<RobotIntent, Num_Shells>& intents,
+    const std::array<MotionSetpoint, Num_Shells>& setpoints) {
+    construct_tx_proto(radioTx, intents, setpoints);
+
     // Get a list of all the IP addresses this packet needs to be sent to
-    for (int robot_idx = 0; robot_idx < packet.robots_size(); robot_idx++) {
-        const Packet::Control& control = packet.robots(robot_idx).control();
-        uint32_t robot_id = packet.robots(robot_idx).uid();
+    for (int robot_idx = 0; robot_idx < radioTx.robots_size(); robot_idx++) {
+        const Packet::Control& control = radioTx.robots(robot_idx).control();
+        uint32_t robot_id = radioTx.robots(robot_idx).uid();
```

Comment:
The whole robot ID being different from index in radioTx array is obe if we are just going to create a `robot` for every single ID. Probably not in scope for this, but it's something to be aware of for future refactoring efforts

Reply:
True. Especially since we're moving towards this type of interface (vector of objects indexed by robot id) I would prefer we fix that in RadioTx. Probably not in this PR, though.